libavcodec/hevc_mp4toannexb_bsf.c

```diff
@@ -155,10 +155,14 @@ static int hevc_mp4toannexb_filter(AVBSFContext *ctx, AVPacket *out)
         }
 
         nalu_type = (bytestream2_peek_byte(&gb) >> 1) & 0x3f;
+        has_vps = (has_vps || nalu_type == HEVC_NAL_VPS);
```

Comment:
vps is optional. most bitstream does not have it.

Reply:
I am aware that vps is optional. This patch is only used to fix some cases, not all cases.

Reply:
but I may remove has_vps in a patch version.

Reply:
removed has_vps in the new version